lms/djangoapps/grades/new/course_grade.py

```diff
@@ -0,0 +1,202 @@
+        children = self.course_structure.get_children(location)
+        earned = 0.0
+        possible = 0.0
+        for child in children:
+            child_earned, child_possible = self.score_for_module(child)
+            earned += child_earned
+            possible += child_possible
+        return earned, possible
+
+    def _compute_letter_grade(self, percentage):
+        """
+        Returns a letter grade as defined in grading_policy (e.g. 'A' 'B' 'C' for 6.002x) or None.
+
+        Arguments
+        - grade_cutoffs is a dictionary mapping a grade to the lowest
+            possible percentage to earn that grade.
+        - percentage is the final percent across all problems in a course
+        """
+
+        letter_grade = None
+        grade_cutoffs = self.course.grade_cutoffs
+
+        # Possible grades, sorted in descending order of score
+        descending_grades = sorted(grade_cutoffs, key=lambda x: grade_cutoffs[x], reverse=True)
+        for possible_grade in descending_grades:
+            if percentage >= grade_cutoffs[possible_grade]:
+                letter_grade = possible_grade
+                break
+
+        return letter_grade
+
+    def _signal_listeners_when_grade_computed(self):
+        """
+        Signal all listeners when grades are computed.
+        """
+        responses = GRADES_UPDATED.send_robust(
+            sender=None,
+            username=self.student.username,
+            grade_summary=self.summary,
+            course_key=self.course.id,
+            deadline=self.course.end
+        )
+
+        for receiver, response in responses:
+            log.info(
+                'Signal fired when student grade is calculated. Receiver: %s. Response: %s',
+                receiver, response
+            )
+
+
+class CourseGradeFactory(object):
+    """
+    Factory class to create Course Grade objects
+    """
+    def __init__(self, student):
+        self.student = student
+
+    def create(self, course):
+        """
+        Returns the CourseGrade object for the given student and course.
+        """
+        course_structure = get_course_blocks(self.student, course.location)
+        return (
+            self._get_saved_grade(course, course_structure) or
+            self._compute_and_update_grade(course, course_structure)
+        )
+
+    def _compute_and_update_grade(self, course, course_structure):
+        """
+        Freshly computes and updates the grade for the student and course.
+        """
+        course_grade = CourseGrade(self.student, course, course_structure)
+        course_grade.compute()
+        return course_grade
+
+    def _get_saved_grade(self, course, course_structure):  # pylint: disable=unused-argument
+        """
+        Returns the saved grade for the given course and student.
+        """
+        if course.enable_subsection_grades_saved:
+            # TODO LATER Retrieve the saved grade for the course, if it exists.
+            pass
```

Comment:
As discussed - feel free to call an arbitrarily-named method here for mocking during unit tests.